Libs/MRML/Core/vtkMRMLPlotNode.h

```diff
@@ -0,0 +1,249 @@
+/*==============================================================================
+
+  Copyright (c) Kapteyn Astronomical Institute
+  University of Groningen, Groningen, Netherlands. All Rights Reserved.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  This file was originally developed by Davide Punzo, Kapteyn Astronomical Institute,
+  and was supported through the European Research Consil grant nr. 291531.
+
+==============================================================================*/
+
+#ifndef __vtkMRMLPlotNode_h
+#define __vtkMRMLPlotNode_h
+
+#include <string>
+#include <vector>
+
+// MRML Includes
+#include "vtkMRMLNode.h"
+
+class vtkMRMLStorageNode;
+class vtkMRMLTableNode;
+
+// VTK Includes
+class vtkColor4ub;
+class vtkPlot;
+class vtkTable;
+
+/// \brief MRML node to represent a Plot object
```

Comment:
Hi @lassoan, sorry for the late reply. I understand the call for a simpler MRML nodes structure, however I still prefer the two nodes structure for the following reasons: 
1) I actually like the double structure because of the information regarding one MRMLPlotNode is stored and always accessible by the scene. For example, if I remove the ID of a MRMLPlotNode from MRMLPlotLayoutNode, I can always readd it anytime. This is useful in the controller widget. See example in this video:
https://www.dropbox.com/s/08lubmg2bx7wcbz/videoPlot1.mkv?dl=0

while in your proposed case, the user needs to recreate the plot or I have to add something like a visibility parameter (of course then the widget needs some work).

2) Having two classes make a clear distinction between the Layout and its properties (e.g., type of the font, size of the font, etc..) and of the properties of the single plot (e.g., width of the line of the plot).

3) having a MRMLPlotLayoutNode with only IDs of MRMLPlotNodes (instead of storing the pointers of vtkplots directly) avoids duplication of memory for creating the same vtkPlot in more "Layout" nodes.

Please let me know. If you really think a simpler MRML infrastructure in this case is much better anyway, then I'll see how to rearrange everything.